paasta_tools/automatic_rollbacks/slack.py

```diff
@@ -295,28 +295,42 @@ def send_initial_slack_message(self):
         self.detail_slack_ts = resp['message']['ts'] if resp and resp['ok'] else None
 
         if resp["ok"] is not True:
-            log.error("Posting detail to slack failed: {}".format(resp["error"]))
+            log.error(f"Posting detail to slack failed: {resp['error']}")
+            if resp['error'] == 'invalid_blocks':
+                log.error(f"Blocks: {detail_blocks!r}")
 
     def update_slack(self):
         summary_blocks = self.get_summary_blocks_for_deployment()
         detail_blocks = self.get_detail_slack_blocks_for_deployment()
-        resp = self.slack_client.api_call(
-            "chat.update",
-            channel=self.slack_channel_id,
-            blocks=summary_blocks,
-            ts=self.slack_ts,
-        )
-        if resp["ok"] is not True:
-            log.error("Posting to slack failed: {}".format(resp["error"]))
 
-        resp = self.slack_client.api_call(
-            "chat.update",
-            channel=self.slack_channel_id,
-            blocks=detail_blocks,
-            ts=self.detail_slack_ts,
-        )
-        if resp["ok"] is not True:
-            log.error("Posting to slack failed: {}".format(resp["error"]))
+        summary_blocks_str = json.dumps(summary_blocks, sort_keys=True)
+        detail_blocks_str = json.dumps(detail_blocks, sort_keys=True)
+
+        if getattr(self, 'old_summary_blocks_str', '') != summary_blocks_str:
```

Comment:
this getattr things is kind weird, why not make this a normal attribute and default it to empty string?

Reply:
Yeah, that's probably better.